pulp_container/app/registry_api.py

```diff
@@ -602,31 +602,31 @@ def mount_blob(self, request, path, repository):
         except models.Blob.DoesNotExist:
             raise BlobNotFound(digest=digest)
 
-        dispatched_task = dispatch(
-            add_and_remove,
-            shared_resources=[version.repository],
-            exclusive_resources=[repository],
-            kwargs={
-                "repository_pk": str(repository.pk),
-                "add_content_units": [str(blob.pk)],
-                "remove_content_units": [],
-            },
+        contains_units = (
+            Q(reserved_resources_record__contains=[get_url(repository)])
```

Comment:
while i do like the this solution, it sems to be sort of more expensive then just putting a string containing all this info into the shared lock in the format of `f"shared:from_repo:{version.repository.pk}:to_repo:{repository.pk}:mount:{blob.digest}`. Thoughts?

Reply:
Yes, I can do it. Also, I was thinking about adding a new field with `db_index=True` to the Task model. The query might be still slow if there will be hundreds of tasks in the database.

Reply:
That field would be used for fast lookups for known tasks. E.g., I would imagine some kind of labels saved along with tasks. Thoughts, @mdellweg?

Reply:
i mostly meant the `get_url` not the query itself

Reply:
We are going to focus on ephemeral blobs first. So, I am adopting the suggested change for now.